### ISSUE TITLE
Option for weighted/unweighted `n.risk`, `n.censor`, etc. with weighted `survfit()` call

### ISSUE THREAD

**Issue by ddsjoberg** [pharmaverse/ggsurvfit#203]
Thank you @therneau for the note! "When there are case weights, do we want the weighted number at risk/events/censored or unweighted?    I lean towards the unweighted for labels, weighted for follow-on computaions, but could argue both sides"

**Comment by VenusSo** [comment]
@ddsjoberg Yes, unweighted is preferred for the plot output. Currently, it's showing weighted with a lot of decimal places. It would be great if the default can be the unweighted counts. Thanks!

**Comment by VenusSo** [comment]
Hi @ddsjoberg, attached is a minimal set of sample data and the resulting KMplot with the `weights` argument turned on in the following minimal function.  Some of the group_1 subjects have <1 weight leading to decimal counts. My current workaround is to round off the n.risk count with the `kmfit$n.risk <- round(kmfit$n.risk)` line in my function. 

```
ggsurv <- function(OSdf, weights=NA) {

  form <- as.formula(Surv(timetoevent, status) ~ drug)

  if(is.na(weights)==TRUE) {
    kmfit <- survfit2(form, data = OSdf)
  } else {
    kmfit <- survfit2(form, data = OSdf, weights = OSdf[ , weights])
  }
  
  # # round the n.risk (for weighted data)
  # kmfit$n.risk <- round(kmfit$n.risk)
  
  plot <- kmfit |>
    ggsurvfit() +
    # add_censor_mark(size = 1, alpha = 0.5) +  # lead to error in `tidyr::uncount()` for weighted data
    add_risktable(
      hjust = 0,
      risktable_stats = c("n.risk"))
  
  return(plot)
}

ggsurv(sampledata, weights="weight")
```

![Image](https://github.com/user-attachments/assets/7df2ec6d-c7d3-40c4-93d5-85a05b670df6)

At a second thought, I think rounding the weighted count is the preferred default since this is how it's handled in the `survminer::ggsurvplot()`.  It will be nice to provide an option to output unweighted count, for example through a  a `fit =`  option in `add_risktable()`. Just a suggestion. Thanks again!

**Comment by ddsjoberg** [comment]
Hi @VenusSo 

You can round the stats in the risktable using `glue::glue()` syntax to inject a rounding function. Please review the `?add_risktable()` help file for an example that calls out weighted models for this particular use case.

You can also report unweighted counts for a weighted model. Here's an example.
https://github.com/pharmaverse/ggsurvfit/issues/235

**Comment by VenusSo** [comment]
@ddsjoberg Ah great! Thanks for pointing that out!
